deployment/terraform/keycloak.go

```diff
@@ -0,0 +1,205 @@
+	url := fmt.Sprintf("http://%s:8080/health", t.output.KeycloakServer.PublicDNS)
+	timeout := time.After(120 * time.Second) // yes, is **that** slow
+	for {
+		resp, err := http.Get(url)
+		if err == nil && resp.StatusCode == http.StatusOK {
+			break
+		}
+		mlog.Info("Service not up yet, waiting...")
+		select {
+		case <-timeout:
+			return errors.New("timeout: keycloak service is not responding")
+		case <-time.After(5 * time.Second):
+		}
+	}
+
+	_, err = sshc.RunCommand(fmt.Sprintf("%s/kcadm.sh config credentials --server http://127.0.0.1:8080 --user %s --password %s --realm master", keycloakBinPath, t.config.ExternalAuthProviderSettings.KeycloakAdminUser, t.config.ExternalAuthProviderSettings.KeycloakAdminPassword))
+	if err != nil {
+		return fmt.Errorf("failed to authenticate keycload admin: %w", err)
+	}
+
+	// Disable SSL requirement on master realm to allow http connections on the web interface
+	_, err = sshc.RunCommand(keycloakBinPath + "/kcadm.sh update realms/master -s sslRequired=NONE")
+	if err != nil {
+		return fmt.Errorf("failed to disable ssl requirement: %w", err)
+	}
+
+	// Disable SSL requirement on the realm set up by the loadtest
+	if t.config.ExternalAuthProviderSettings.KeycloakRealmName != "" && t.config.ExternalAuthProviderSettings.KeycloakRealmName != "master" {
+		_, err = sshc.RunCommand(fmt.Sprintf(keycloakBinPath+"/kcadm.sh update realms/%s -s sslRequired=NONE", t.config.ExternalAuthProviderSettings.KeycloakRealmName))
+		if err != nil {
+			return fmt.Errorf("failed to disable ssl requirement: %w", err)
+		}
+	}
+
+	// Populate users
+	if t.config.ExternalAuthProviderSettings.GenerateUsersCount > 0 {
+		if err := t.populateKeycloakUsers(sshc); err != nil {
+			return fmt.Errorf("failed to populate keycloak users: %w", err)
+		}
+
+		mlog.Info("Overriding the users file path with the generated one from keycloak")
+		t.config.UsersFilePath = t.getAsset("keycloak-users.txt")
+	}
+
+	mlog.Info("Keycloak configured")
+
+	return nil
+}
+
+func (t *Terraform) populateKeycloakUsers(sshc *ssh.Client) error {
+	keycloakBinPath := "/opt/keycloak/keycloak-" + t.config.ExternalAuthProviderSettings.KeycloakVersion + "/bin"
+	usersTxtPath := t.getAsset("keycloak-users.txt")
+
+	// Check if users file exists. Prevents from creating users multiple times.
+	if _, err := os.Stat(usersTxtPath); err == nil || os.IsExist(err) {
+		return nil
+	}
+
+	mlog.Info("Populating keycloak with users", mlog.String("users_file", usersTxtPath), mlog.Int("users_count", t.config.ExternalAuthProviderSettings.GenerateUsersCount))
+
+	// Open users.txt file
+	handler, err := os.OpenFile(usersTxtPath, os.O_CREATE|os.O_WRONLY, 0666)
+	if err != nil {
+		return fmt.Errorf("failed to open keycloak users file: %w", err)
+	}
+	defer handler.Close()
+
+	// Create users
+	for i := 1; i <= t.config.ExternalAuthProviderSettings.GenerateUsersCount; i++ {
+		username := fmt.Sprintf("keycloak-auto-%06d", i) // this is the password too
```

Comment:
Nit to make the code below a bit more clear
```suggestion
		username := fmt.Sprintf("keycloak-auto-%06d", i)
		password := username // we just use the same string for both
```